### ISSUE TITLE
Sprite clipping in geometry shader a bit too aggressive

### ISSUE THREAD

**Issue by einarf** [pythonarcade/arcade#859]
Looks like the sprite culling the geometry shader is a bit inaccurate for very wide/high sprites. I think this is due to to distance being calculated in normalized device coordinates and doesn't take aspect ratio into account.

Dims of sprite
```py
hit_box_points = [
    [159.116175365299, 0.6560658013133782],
    [159.496175365299, -1.1039341986866198],
    [439.896175365299, 59.176065801313385],
    [439.516175365299, 60.936065801313376]
]

height = 286.8
width = 1.8
```

Origin : https://discord.com/channels/458662222697070613/766795048049442836/824612586766467122

https://user-images.githubusercontent.com/5418180/116282837-8c3c8000-a78b-11eb-8b1d-9c9a4ebabe96.mp4

## TODO

- [ ] Test sprite clipping with very tall/long sprites rotated 45, 90, 135, 180, 225, 270, 315 degrees

**Comment by einarf** [comment]
I've managed to reproduce this with the following example. Drag the geometry around and see it dissappearing before leaving the screen.

```py
import PIL
import arcade


class GeoCullingTest(arcade.Window):

    def __init__(self):
        super().__init__(800, 400, "Cull test")
        self.proj = self.ctx.projection_2d
        self.texture = arcade.Texture("weird_texture", image=PIL.Image.new("RGBA", (1024, 2), (255, 255, 255, 255)))

        self.spritelist = arcade.SpriteList()
        for i in range(0, 360, 36):
            self.sprite = arcade.Sprite(texture=self.texture, center_x=400, center_y=300)
            self.sprite.angle = i
            self.spritelist.append(self.sprite)

    def on_draw(self):
        self.clear()
        self.ctx.projection_2d = self.proj
        self.spritelist.draw()

    def on_mouse_drag(self, x: float, y: float, dx: float, dy: float, buttons: int, modifiers: int):
        self.proj = (
            self.proj[0] - dx,
            self.proj[1] - dx,
            self.proj[2] - dy,
            self.proj[3] - dy,
        )


window = GeoCullingTest()
arcade.run()
```

**Comment by einarf** [comment]
Fixed in cd4fc12

**Comment by Cleptomania** [comment]
Released with 2.6.0
